File: python/greedy/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        finalanswer = float('-inf')
        current = float('-inf')
        for i in nums:
            current = max(current + i, i)
            finalanswer = max(finalanswer, current)

        return finalanswer

File: c++/greedy/maximum-subarray.cc
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int answer = std::numeric_limits<int>::min();
        int curr = 0;
        for (int num : nums) {
            curr = std::max(num, curr + num);
            answer = std::max(answer, curr);
        }
        return answer;
    }
};
